nodejs/eks/cluster.ts

```diff
@@ -866,43 +887,63 @@ export class Cluster extends pulumi.ComponentResource {
+                tags: pulumi.all([
+                    args.tags,
+                    args.nodeSecurityGroupTags,
+                ]).apply(([tags, nodeSecurityGroupTags]) => (<aws.Tags>{
+                    ...nodeSecurityGroupTags,
+                    ...tags,
+                })),
+            }, this);
+            core.nodeSecurityGroup = this.nodeSecurityGroup;
+        }
 
         const configDeps = [core.kubeconfig];
         if (!args.skipDefaultNodeGroup) {
             // Create the worker node group and grant the workers access to the API server.
-            this.defaultNodeGroup = createNodeGroup(name, {
-                cluster: core,
-                nodeSubnetIds: args.nodeSubnetIds,
-                nodeAssociatePublicIpAddress: args.nodeAssociatePublicIpAddress,
-                nodeSecurityGroup: this.nodeSecurityGroup,
-                clusterIngressRule: this.eksClusterIngressRule,
-                instanceType: args.instanceType,
-                nodePublicKey: args.nodePublicKey,
-                nodeRootVolumeSize: args.nodeRootVolumeSize,
-                nodeUserData: args.nodeUserData,
-                minSize: args.minSize,
-                maxSize: args.maxSize,
-                desiredCapacity: args.desiredCapacity,
-                amiId: args.nodeAmiId,
-                version: args.version,
-                instanceProfile: core.instanceProfile,
-            }, this);
-            this.nodeSecurityGroup = this.defaultNodeGroup.nodeSecurityGroup;
-            configDeps.push(this.defaultNodeGroup.cfnStack.id);
+            if (args.nodeGroupOptions !== undefined) {
+                this.defaultNodeGroup = createNodeGroup(name, {
+                    ...args.nodeGroupOptions,
+                    cluster: core,
+                    nodeSecurityGroup: this.nodeSecurityGroup,
+                    clusterIngressRule: this.eksClusterIngressRule,
+                }, this);
+            } else {
+                this.defaultNodeGroup = createNodeGroup(name, {
+                    cluster: core,
+                    nodeSubnetIds: args.nodeSubnetIds,
+                    nodeAssociatePublicIpAddress: args.nodeAssociatePublicIpAddress,
+                    nodeSecurityGroup: this.nodeSecurityGroup,
+                    clusterIngressRule: this.eksClusterIngressRule,
+                    instanceType: args.instanceType,
+                    nodePublicKey: args.nodePublicKey,
+                    nodeRootVolumeSize: args.nodeRootVolumeSize,
+                    nodeUserData: args.nodeUserData,
+                    minSize: args.minSize,
+                    maxSize: args.maxSize,
+                    desiredCapacity: args.desiredCapacity,
+                    amiId: args.nodeAmiId,
+                    version: args.version,
+                    instanceProfile: core.instanceProfile,
+                }, this);
+                this.nodeSecurityGroup = this.defaultNodeGroup.nodeSecurityGroup;
+            }
```

Comment:
> It feels like this might be cleaner and more likely to keep correct if the if/else just constructed the options, and there was a single call to createNodeGroup outside that.

Sounds good.

> For example, are you sure we don't need this.nodeSecurityGroup = in the first branch above? That looks like a bug.

`this.nodeSecurityGroup` gets set prior to this branch exec'ing per the new conditional branch, and then ref'd.

https://github.com/pulumi/pulumi-eks/blob/2643dfbf3b1e6b4ae28fca74c387fdbc56694816/nodejs/eks/cluster.ts#L890-L896

Reply:
Done